docs/design.md

```diff
@@ -0,0 +1,110 @@
+ * Support OCI image signature verification
+ * Support configurable security policies for allowed OCI images
+ * Support seamlessly integration with Kata Agent and Attestation Agent
+ * Provide a command-line utility for debug
+ * Explore image decryption and decompression acceleration technologies
+ * Explore image on demand loading technologies
+ * Explore technologies to share image data among containers/pods with confidentiality and integrity.
+
+
+### Non-Goals
+ * Image push, content discovery/management operations for remote registries
+ * Image encryption and signing
+ * Image export to local storage
+ * Image import from other service daemon or other local storage format
+
+
+### Existing Solutions
+ * [containers-image-proxy-rs](https://github.com/containers/containers-image-proxy-rs): a rust bindings for skopeo,
+   which doesn't meet our requirements of small TCB and lightweight.
+ * [oci-distribution](https://github.com/krustlet/oci-distribution): a crate from
+   [krustlet](https://github.com/krustlet/krustlet) to run wasm application with K8S. Since wasm is platform agnostic
+   and small code size, `oci-distribution` does not support manifest lists for different platforms and internal data
+   struct is not friend to generic container workload.
+ * [oci-registry-client](https://github.com/ecarrara/oci-registry-client): a very simple client for OCI image registries
+   with limited features and latest commit is one and half years ago.
+
+Above projects focus on image distribution and do not support image unpack and storage management features.
+
+
+## Architecture
+The following diagram demonstrates the overall architecture and how to
+integrate with Kata Agent:
+![Architecture](images/architecture.png)
+
+Diagram source file: [architecture.pptx](images/architecture.pptx)
+
+`image-rs` provides the API which mainly cover image pulling and management.
+
+For image pulling part, it will follow OCI distribution and image spec to
+interact with remote registries to download container image metadata
+and blob data, these info will be consumed by the image management module.
+After container image is downloaded, `image-rs` will talk with remote
+attestation service to verify the image signature and decrypt the image
+if needed. After image is unpacked, the image format module will handle
+the format conversion between oci v1 and docker v2.
+
+The image pulling module cooperates with the
+[Attestation-agent](https://github.com/confidential-containers/attestation-agent)
+to fetch secrets for decryption/verification from the remote attestation and key management server.
+
+Snapshot module will prepare the rootfs together with the container
+configuration and metadata, the bundles can be generated for the
+runtime service to consume. snapshot module also can support different
+snapshotters which can support image on demand pulling.
+
+
+Due to data decryption and decompression can be time consuming, image-rs will
```

Comment:
`As container image layers decryption and decompression can be time consuming, ...`

Reply:
Thanks, will update.